### DISCUSSION TITLE
target(test): toolchain not found!

### DISCUSSION THREAD

**Discussion by Kev1n3zz** [xmake-io/xmake#1996] [Q&A]
作者您好，小白初装xmake，系统内有预先安装的MinGW

在生成test文件后使用
xmake run test 产生如下报错：
checking for platform ... windows
checking for architecture ... x64
checking for Microsoft Visual Studio (x64) version ... no
error: target(test): toolchain not found!

请问我一定要下载VS才能解决吗？我是否可以在VSCode中解决这个问题？

**Comment by waruqi** [comment]
可以切到 mingw 平台编译

xmake f -p mingw --mingw=/xxx/mingw/sdk 
xmake

**Comment by Kev1n3zz** [comment]
我按照您的方法xmake f --mingw=/xxx/mingw/sdk不奏效
使用：
xmake f --mingw=/xxx/mingw
可以通过

xmake run test 后在build文件中产生了一个包含.exe的MinGW文件
请问这里是否是我使用MinGW平台编译了win上的应用？

**Comment by waruqi** [comment]
> 我按照您的方法xmake f --mingw=/xxx/mingw/sdk不奏效 使用： xmake f --mingw=/xxx/mingw 可以通过

路径我随便写的，得指定你自己的 mingw 路径，并且切换到 -p mingw 平台编译

> 
> xmake run test 后在build文件中产生了一个包含.exe的MinGW文件 请问这里是否是我使用MinGW平台编译了win上的应用？

自己 -vD 看下编译详情，自己确认下
